boards/nxp/ucans32k146/src/can_boot.c

```diff
@@ -75,8 +75,32 @@ __EXPORT void s32k1xx_board_initialize(void)
 	s32k1xx_pinconfig(BOARD_REVISION_DETECT_PIN);
 
 	if (s32k1xx_gpioread(BOARD_REVISION_DETECT_PIN)) {
-		/* STB high -> active CAN phy */
-		s32k1xx_pinconfig(PIN_CAN0_STB  | GPIO_OUTPUT_ONE);
+		/* Config Pins to do CAN tranceiver HW selftest */
+		s32k1xx_pinconfig(PIN_CAN0_ERRN);
+		s32k1xx_pinconfig(PIN_CAN0_STB);
+		s32k1xx_pinconfig(PIN_CAN0_EN);
+
+		/* EN high & STB high -> normal mode */
+		s32k1xx_gpiowrite(PIN_CAN0_STB, 1);
+		s32k1xx_gpiowrite(PIN_CAN0_EN, 1);
+		up_udelay(3000); // Wait for startup to normal mode
+
+		/* EN low & STB high -> listen only mode */
+		s32k1xx_gpiowrite(PIN_CAN0_STB, 1);
+		s32k1xx_gpiowrite(PIN_CAN0_EN, 0);
+		up_udelay(100); // t moch ERRN_N
+
+		/* Check for HW err and wait untill ERR has been cleared */
+		while (!s32k1xx_gpioread(PIN_CAN0_ERRN)) {
+			board_indicate(hardware_failure);
```

Comment:
Is the LED's rate constant? If the LED is on timer, this can come out of the loop.

Reply:
My scope measures a constant 10Hz. 
I can go out of the loop indeed if the short is temporary.